include/deal.II/particles/particle_handler.h

```diff
@@ -0,0 +1,507 @@
+     * pointer that points to a data block of size size_callback() in which
+     * additional data was stored by the store_callback function. The
+     * function is expected to return a void pointer pointing to a position
+     * right after its data block.
+     */
+    void
+    register_additional_store_load_functions(const std::function<std::size_t ()> &size_callback,
+                                             const std::function<void *(const particle_iterator &,
+                                                 void *)> &store_callback,
+                                             const std::function<const void *(const particle_iterator &,
+                                                 const void *)> &load_callback);
+
+    /**
+     * Return the total number of particles that were managed by this class
+     * the last time the update_cached_numbers() function was called.
+     * The actual number of particles may have changed since then if
+     * particles have been added or removed.
+     *
+     * @return Total number of particles in simulation.
+     */
+    types::particle_index n_global_particles() const;
+
+    /**
+     * Return the maximum number of particles per cell the last
+     * time the update_cached_numbers() function was called.
+     *
+     * @return Maximum number of particles in one cell in simulation.
+     */
+    types::particle_index n_global_max_particles_per_cell() const;
+
+    /**
+     * Return the number of particles in the local part of the
+     * triangulation.
+     */
+    types::particle_index n_locally_owned_particles() const;
+
+    /**
+     * Return the next free particle index in the global set
+     * of particles the last
+     * time the update_cached_numbers() function was called.
+     */
+    types::particle_index get_next_free_particle_index() const;
+
+    /**
+     * Return the number of properties each particle has.
+     */
+    unsigned int n_properties_per_particle() const;
+
+    /**
+     * Return a reference to the property pool that owns all particle
+     * properties, and organizes them physically.
+     */
+    PropertyPool &
+    get_property_pool() const;
+
+    /**
+     * Return the number of particles in the given cell.
+     */
+    unsigned int
+    n_particles_in_cell(const typename Triangulation<dim,spacedim>::active_cell_iterator &cell) const;
+
+    /**
+     * Finds the cells containing each particle for all locally owned
+     * particles. If particles moved out of the local subdomain
+     * they will be sent to their new process and inserted there.
+     * After this function call every particle is either on its current
+     * process and in its current cell, or deleted (if it could not find
+     * its new process or cell).
+     */
+    void
+    sort_particles_into_subdomains_and_cells();
+
+
+#ifdef DEAL_II_WITH_MPI
+    /**
+     * Exchanges all particles that live in cells that are ghost cells to
```

Comment:
I would prefer if the declaration was always available, and only the *body* of the implementation is commented out when not using MPI. This way, the function just does nothing, but an MPI code can also be compiled without using MPI.